README.md

```diff
@@ -500,6 +500,75 @@ All non-alpha releases derive from tagged commits on `master`. The tag defines t
 - after the push has resulted in a successful build, click through from the commit on github thru to the Azure Pipelines build state and verify _all_ artifacts bear the correct version suffix (if the tags were not pushed alongside the commit, they can be wrong). Then, and only then, do the Release (which will upload to nuget.org using a nuget API key that has upload permissions for the packages)
 - _When adding new packages_: For safety, the NuGet API Key used by the Azure DevOps Releases step can only upload new versions of existing packages. As a result, the first version of any new package needs to be manually uploaded out of band. (then invite jet.com to become owner so subsequent releases can do an automated upload [after the request has been (manually) accepted])
 
+## Access Strategies
+
+An Access Strategy defines any optimizations regarding how one arrives at a State of an Aggregate based on the Events stored in a Stream in a Store.
+
+The specifics of an Access Strategy depend on what makes sense for a given Store, i.e. `Equinox.Cosmos` necessarily has a significantly different set of strategies than `Equinox.EventStore` (although there is an intersection).
+
+Access Strategies only affect performance; you should still be able to infer the state of the aggregate based on the `fold` of all the `events` ever written on top of an `initial` state
+
+NOTE: its not important to select a strategy until you've actually actually modelled your aggregate, see [what if I change my access strategy](#changing-access-strategy)
+
+### `Equinox.Cosmos.AccessStrategy`
+
+TL;DR `Equinox.Cosmos`: (see also: [the storage model](DOCUMENTATION.md#Cosmos-Storage-Model) for a deep dive, and [glossary, below the table](#access-strategy-glossary) for definition of terms)
+- keeps all the events for a stream in a single single [CosmosDB _logical partition_](https://docs.microsoft.com/en-gb/azure/cosmos-db/partition-data)
+- always has a special 'index' document (we term it the `Tip` document), per logical partition/stream that is accessible via an efficient _point read_ (it always has a CosmosDB `id` value of `"-1"`)
+- Events are stored in immutable documents in the logical partition, the `Tip` document is the only document that's ever updated. 
+- all writes touch the `Tip`, so we have a natural way to invalidate any cached State for a given Stream - we retain the `_etag` of the `Tip` document, and updates (or consistent reads) are contingent on it not having changed
```

Comment:
Just for my own knowledge. Does this `_etag` help with version check when writing an event? Do I need to explicitly compare the state I used for decision making and the current state when I am about the write the event to make sure they are the same before I persist anything?

Reply:
Based on the next statement, looks like the Tip document doesn;t help with race condition of writes and it will just take them one by one. Is this right?

Reply:
the etag check comes into play for every write
writes can only be triggered by returning an event from interpret/decide
so yes, having the decide/interpret maintain enough state to make that dedup decision is an important way to reduce load, and hence RU consumption and/or contention under load (when it matters!)
And yes, the Tip document means we can quickly spot a conflict, but we resolve only be responding with the tip content and letting Transact then retry by re-running interpret/decide (hopefully if it is redundant, that will then end the cycle without a further write)
In eventstore, the EventId can be used to trigger dedup on the server - we could implement something like that in the stored proc but there are no such plans, mainly as I'm not aware of any use cases we have that dont have better solution
Do you feel this maps onto a sentence that would have helped you be certain?

Reply:
Added
> - The `interpret`/`decide` function is expected to deduplicate writes by not producing `events` if the `state` implies such updates would be redundant. Equinox does not have any internal mechanism to deduplicate events, thus having correct deduplication is the key to reducing round-trips and hence minimizing RU consumption (and the adverse effects that the retry cycles due to contention have, which will most likely arise when load is at its highest)

3 below this

Reply:
This is helpful!

Reply:
thanks for prompting it (but others wont get to learn by working it out for themselves 😈 )